vllm/worker/model_runner.py

```diff
@@ -1551,6 +1551,21 @@ def execute_model(
 
         # Compute the logits in the last pipeline stage.
         if not get_pp_group().is_last_rank:
+            if (self.is_driver_worker
+                    and hidden_or_intermediate_states is not None
+                    and isinstance(hidden_or_intermediate_states,
+                                   IntermediateTensors)
+                    and self.observability_config is not None
+                    and self.observability_config.collect_model_forward_time):
+                model_forward_end.synchronize()
+                model_forward_time = model_forward_start.elapsed_time(
+                    model_forward_end)
+                orig_model_forward_time = 0.0
+                if intermediate_tensors is not None:
+                    orig_model_forward_time = intermediate_tensors.tensors.get(
+                        "model_forward_time", torch.tensor(0.0)).item()
```

Comment:
why do we store this to tensor? any way to just use cpu data?

Reply:
as far as I can tell, the only thing passed from the pipeline workers is the IntermediateTensors in serialized form. Hence added it to that. Is there a wrapper object of some form that holds these ?

Reply:
Done. It looks like the worker serializes a Dict[Str, Any], so it can serialize floats too.